diff --git a/CMakeLists.txt b/CMakeLists.txt
index 10164bc0..cd87e790 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -68,4 +68,4 @@ ) # The exact amount of indentation used in the line(s) above is intentional

-option(BINARY_ICONS_RESOURCE "Install Qt binary resource files containing breeze icons (breeze-icons.rcc, breeze-icons-dark.rcc)" OFF)
+option(BINARY_ICONS_RESOURCE "Install Qt binary resource files containing breeze icons (breeze-icons.rcc, breeze-icons-dark.rcc)" ON)
 option(SKIP_INSTALL_ICONS "Skip installing the icons files" OFF)
 option(ICONS_LIBRARY "Install a library including the breeze & breeze dark icons as resources & functions to use them. This is intended for self contained deployments and should generally not be enabled for systems that have the concept of icon themes like Linux distributions/*BSD" OFF)
