diff --git a/src/sycoca/ksycoca.cpp b/src/sycoca/ksycoca.cpp
index 05cb32c1..b6b3da78 100644
--- a/src/sycoca/ksycoca.cpp
+++ b/src/sycoca/ksycoca.cpp
@@ -683,22 +683,9 @@ bool KSycocaPrivate::buildSycoca()
 
 QString KSycoca::absoluteFilePath()
 {
-    const QStringList paths = QStandardPaths::standardLocations(QStandardPaths::GenericDataLocation);
-    QString suffix = QLatin1Char('_') + QLocale().bcp47Name();
-
-    const QByteArray ksycoca_env = qgetenv("KDESYCOCA");
-    if (ksycoca_env.isEmpty()) {
-        const QByteArray pathHash = QCryptographicHash::hash(paths.join(QLatin1Char(':')).toUtf8(), QCryptographicHash::Sha1);
-        suffix += QLatin1Char('_') + QString::fromLatin1(pathHash.toBase64());
-        suffix.replace(QLatin1Char('/'), QLatin1Char('_'));
-#ifdef Q_OS_WIN
-        suffix.replace(QLatin1Char(':'), QLatin1Char('_'));
-#endif
-        const QString fileName = QLatin1String("ksycoca6") + suffix;
-        return QStandardPaths::writableLocation(QStandardPaths::GenericCacheLocation) + QLatin1Char('/') + fileName;
-    } else {
-        return QFile::decodeName(ksycoca_env);
-    }
+    return QStandardPaths::writableLocation(QStandardPaths::GenericCacheLocation) +
+                                            QLatin1Char('/')                      +
+                                            QLatin1String("ksycoca_digikam");
 }
 
 QStringList KSycoca::allResourceDirs()
