diff --git a/CMakeLists.txt b/CMakeLists.txt
index 533815d2..1f384297 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -94,35 +94,35 @@ endif()
 get_property(SHARED_LIBS_SUPPORTED GLOBAL PROPERTY TARGET_SUPPORTS_SHARED_LIBS)
 option(BUILD_SHARED_LIBS "Build shared libraries instead of static ones" ${SHARED_LIBS_SUPPORTED})
 
-set(JPEGXL_ENABLE_FUZZERS ${ENABLE_FUZZERS_DEFAULT} CACHE BOOL
+set(JPEGXL_ENABLE_FUZZERS false CACHE BOOL
     "Build JPEGXL fuzzer targets.")
 set(JPEGXL_ENABLE_DEVTOOLS false CACHE BOOL
     "Build JPEGXL developer tools.")
-set(JPEGXL_ENABLE_TOOLS true CACHE BOOL
+set(JPEGXL_ENABLE_TOOLS false CACHE BOOL
     "Build JPEGXL user tools: cjxl and djxl.")
-set(JPEGXL_ENABLE_DOXYGEN true CACHE BOOL
+set(JPEGXL_ENABLE_DOXYGEN false CACHE BOOL
     "Generate C API documentation using Doxygen.")
-set(JPEGXL_ENABLE_MANPAGES true CACHE BOOL
+set(JPEGXL_ENABLE_MANPAGES false CACHE BOOL
     "Build and install man pages for the command-line tools.")
-set(JPEGXL_ENABLE_BENCHMARK true CACHE BOOL
+set(JPEGXL_ENABLE_BENCHMARK false CACHE BOOL
     "Build JPEGXL benchmark tools.")
-set(JPEGXL_ENABLE_EXAMPLES true CACHE BOOL
+set(JPEGXL_ENABLE_EXAMPLES false CACHE BOOL
     "Build JPEGXL library usage examples.")
-set(JPEGXL_BUNDLE_LIBPNG ${BUNDLE_LIBPNG_DEFAULT} CACHE BOOL
+set(JPEGXL_BUNDLE_LIBPNG false CACHE BOOL
     "Build libpng from source and link it statically.")
-set(JPEGXL_ENABLE_JNI true CACHE BOOL
+set(JPEGXL_ENABLE_JNI false CACHE BOOL
     "Build JPEGXL JNI Java wrapper, if Java dependencies are installed.")
-set(JPEGXL_ENABLE_SJPEG true CACHE BOOL
+set(JPEGXL_ENABLE_SJPEG false CACHE BOOL
     "Build JPEGXL with support for encoding with sjpeg.")
-set(JPEGXL_ENABLE_OPENEXR true CACHE BOOL
+set(JPEGXL_ENABLE_OPENEXR false CACHE BOOL
     "Build JPEGXL with support for OpenEXR if available.")
-set(JPEGXL_ENABLE_SKCMS true CACHE BOOL
+set(JPEGXL_ENABLE_SKCMS false CACHE BOOL
     "Build with skcms instead of lcms2.")
-set(JPEGXL_BUNDLE_SKCMS true CACHE BOOL
+set(JPEGXL_BUNDLE_SKCMS false CACHE BOOL
     "When building with skcms, bundle it into libjxl.a.")
 set(JPEGXL_ENABLE_VIEWERS false CACHE BOOL
     "Build JPEGXL viewer tools for evaluation.")
-set(JPEGXL_ENABLE_TCMALLOC ${ENABLE_TCMALLOC_DEFAULT} CACHE BOOL
+set(JPEGXL_ENABLE_TCMALLOC false CACHE BOOL
     "Build JPEGXL using gperftools (tcmalloc) allocator.")
 set(JPEGXL_ENABLE_PLUGINS false CACHE BOOL
     "Build third-party plugins to support JPEG XL in other applications.")
@@ -138,7 +138,7 @@ set(JPEGXL_ENABLE_TRANSCODE_JPEG true CACHE BOOL
  (default enabled)")
 set(JPEGXL_STATIC false CACHE BOOL
     "Build tools as static binaries.")
-set(JPEGXL_WARNINGS_AS_ERRORS ${WARNINGS_AS_ERRORS_DEFAULT} CACHE BOOL
+set(JPEGXL_WARNINGS_AS_ERRORS false CACHE BOOL
     "Treat warnings as errors during compilation.")
 set(JPEGXL_DEP_LICENSE_DIR "" CACHE STRING
     "Directory where to search for system dependencies \"copyright\" files.")
@@ -469,4 +469,4 @@ add_subdirectory(plugins)
 endif ()
 
 # Binary tools
-add_subdirectory(tools)
+#add_subdirectory(tools)
