diff --git a/CMakeLists.txt b/CMakeLists.txt
index 58ade87..e64558f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -63,17 +63,17 @@ if (WITH_X265)
     find_package(X265)
 endif ()
 
-option(WITH_AOM "Build aom encoder/decoder" ON)
+option(WITH_AOM "Build aom encoder/decoder" OFF)
 if (WITH_AOM)
     find_package(LibAOM)
 endif()
 
-option(WITH_RAV1E "Build rav1e encoder" ON)
+option(WITH_RAV1E "Build rav1e encoder" OFF)
 if (WITH_RAV1E)
     find_package(Rav1e)
 endif ()
 
-option(WITH_DAV1D "Build dav1d decoder" ON)
+option(WITH_DAV1D "Build dav1d decoder" OFF)
 if (WITH_DAV1D)
     find_package(Dav1d)
 endif ()
@@ -159,10 +159,10 @@ configure_file(libheif.pc.in ${CMAKE_CURRENT_BINARY_DIR}/libheif.pc @ONLY)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libheif.pc
         DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 
-option(WITH_EXAMPLES "Build examples" ON)
+option(WITH_EXAMPLES "Build examples" OFF)
 
 if(WITH_EXAMPLES)
     add_subdirectory (examples)
 endif()
 add_subdirectory (libheif)
-add_subdirectory (gdk-pixbuf)
+#add_subdirectory (gdk-pixbuf)
