diff --git a/modules/ECMQmlModule6.cmake b/modules/ECMQmlModule6.cmake
index 198f7449..a953e815 100644
--- a/modules/ECMQmlModule6.cmake
+++ b/modules/ECMQmlModule6.cmake
@@ -21,7 +21,7 @@ if (NOT QT_QML_OUTPUT_DIRECTORY)
 endif()
 
 # Stop warning about a changed import prefix.
-qt6_policy(SET QTP0001 NEW)
+#qt6_policy(SET QTP0001 NEW)
 
 function(ecm_add_qml_module ARG_TARGET)
     cmake_parse_arguments(PARSE_ARGV 1 ARG "NO_PLUGIN;QT_NO_PLUGIN;GENERATE_PLUGIN_SOURCE" "URI;VERSION;CLASSNAME" "")
