diff --git a/src/imageformats/exr.cpp b/src/imageformats/exr.cpp
index c3c57a3..69e168b 100644
--- a/src/imageformats/exr.cpp
+++ b/src/imageformats/exr.cpp
@@ -190,11 +190,11 @@ bool EXRHandler::read(QImage *outImage)
             if (auto utcOffset = h.findTypedAttribute<Imf::FloatAttribute>("utcOffset")) {
                 off = utcOffset->value();
             }
-            auto dateTime = QDateTime::fromString(QString::fromStdString(capDate->value()), QStringLiteral("yyyy:MM:dd HH:mm:ss"));
+/*            auto dateTime = QDateTime::fromString(QString::fromStdString(capDate->value()), QStringLiteral("yyyy:MM:dd HH:mm:ss"));
             if (dateTime.isValid()) {
                 dateTime.setTimeZone(QTimeZone::fromSecondsAheadOfUtc(off));
                 image.setText(QStringLiteral("Date"), dateTime.toString(Qt::ISODate));
-            }
+            }*/
         }
         if (auto xDensity = h.findTypedAttribute<Imf::FloatAttribute>("xDensity")) {
             float par = 1;
