diff --git a/CMakeLists.txt b/CMakeLists.txt
index b339f899..28c0524c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,7 +20,7 @@ include (${CMAKE_ROOT}/Modules/CheckIncludeFile.cmake)
 include(GNUInstallDirs)
 include(CheckFunctionExists)
 
-find_package(SDL)
+#find_package(SDL)
 find_package(Threads REQUIRED)
 
 CHECK_INCLUDE_FILE(malloc.h HAVE_MALLOC_H)
@@ -64,5 +64,5 @@ if(MSVC)
 endif()
 
 add_subdirectory (libde265)
-add_subdirectory (dec265)
-add_subdirectory (enc265)
+#add_subdirectory (dec265)
+#add_subdirectory (enc265)
