diff --git a/CMakeLists.txt b/CMakeLists.txt
index 70b334818..19d56875a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -49,7 +49,7 @@ ecm_setup_version(PROJECT VARIABLE_PREFIX KCALENDARCORE
 ########### Find packages ###########
 find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Core Gui)
 
-set(LibIcal_MIN_VERSION "3.0")
+#set(LibIcal_MIN_VERSION "3.0")
 find_package(LibIcal ${LibIcal_MIN_VERSION})
 set_package_properties(LibIcal PROPERTIES TYPE REQUIRED)
 
