diff --git a/CMakeLists.txt b/CMakeLists.txt
index cbdf54e20..44a2d1b23 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,11 +47,11 @@ ecm_setup_version(PROJECT VARIABLE_PREFIX KCALENDARCORE
 ########### Find packages ###########
 find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Core Gui)
 
-set(LibIcal_MIN_VERSION "3.0")
+#set(LibIcal_MIN_VERSION "3.0")
 find_package(LibIcal ${LibIcal_MIN_VERSION})
 set_package_properties(LibIcal PROPERTIES TYPE REQUIRED)
-set(HAVE_ICAL_3 TRUE)
-add_definitions(-DUSE_ICAL_3)
+#set(HAVE_ICAL_3 TRUE)
+#add_definitions(-DUSE_ICAL_3)
 
 ########### CMake Config Files ###########
 set(CMAKECONFIG_INSTALL_DIR "${KDE_INSTALL_CMAKEPACKAGEDIR}/KF5CalendarCore")
