diff --git a/src/lib/util/ksignalhandler.cpp b/src/lib/util/ksignalhandler.cpp
index 1a644805..bafb8c69 100644
--- a/src/lib/util/ksignalhandler.cpp
+++ b/src/lib/util/ksignalhandler.cpp
@@ -34,7 +34,8 @@ KSignalHandler::KSignalHandler()
 {
     d->q = this;
 #ifndef Q_OS_WIN
-    if (::socketpair(AF_UNIX, SOCK_STREAM | SOCK_CLOEXEC, 0, KSignalHandlerPrivate::signalFd)) {
+    int fd = ::socketpair(AF_UNIX, SOCK_STREAM, 0, KSignalHandlerPrivate::signalFd);
+    if (fcntl(fd, O_CLOEXEC) == -1) {
         qCWarning(KCOREADDONS_DEBUG) << "Couldn't create a socketpair";
         return;
     }
