diff --git a/qtmultimedia/src/multimedia/configure.cmake b/qtmultimedia/src/multimedia/configure.cmake
index 609806507..4169ab238 100644
--- a/qtmultimedia/src/multimedia/configure.cmake
+++ b/qtmultimedia/src/multimedia/configure.cmake
@@ -40,10 +40,10 @@ qt_config_compile_test("gpu_vivante"
                    PROJECT_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../config.tests/gpu_vivante"
 )
 
-qt_config_compile_test("linux_v4l"
-                   LABEL "Video for Linux"
-                   PROJECT_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../config.tests/linux_v4l"
-)
+#qt_config_compile_test("linux_v4l"
+#                   LABEL "Video for Linux"
+#                   PROJECT_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../config.tests/linux_v4l"
+#)
 
 qt_config_compile_test("wmsdk"
                    LABEL "wmsdk.h"
