diff --git a/CMakeLists.txt b/CMakeLists.txt
index 08f6add..c8a725a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,15 +23,15 @@ set(libjbig_src
   "${CMAKE_CURRENT_SOURCE_DIR}/libjbig/jbig.c"
   "${CMAKE_CURRENT_SOURCE_DIR}/libjbig/jbig_ar.c"
 )
-add_library(libjbig ${libjbig_src})
+add_library(jbig ${libjbig_src})
 
 set(libjbig85_src 
   "${CMAKE_CURRENT_SOURCE_DIR}/libjbig/jbig85.c"
   "${CMAKE_CURRENT_SOURCE_DIR}/libjbig/jbig_ar.c"
 )
-add_library(libjbig85 ${libjbig85_src})
-install(TARGETS libjbig RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
-install(TARGETS libjbig85 RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+add_library(jbig85 ${libjbig85_src})
+install(TARGETS jbig RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+install(TARGETS jbig85 RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
 install(FILES
   "${CMAKE_CURRENT_SOURCE_DIR}/libjbig/jbig.h"
   "${CMAKE_CURRENT_SOURCE_DIR}/libjbig/jbig_ar.h"
@@ -42,36 +42,36 @@ install(FILES
 # Programs
 ########################################
 
-add_executable (tstcodec "${CMAKE_CURRENT_SOURCE_DIR}/libjbig/tstcodec.c")
-set_target_properties(tstcodec PROPERTIES DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX})
-target_link_libraries (tstcodec libjbig$<$<CONFIG:Debug>:d>)
-install(TARGETS tstcodec RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+#add_executable (tstcodec "${CMAKE_CURRENT_SOURCE_DIR}/libjbig/tstcodec.c")
+#set_target_properties(tstcodec PROPERTIES DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX})
+#target_link_libraries (tstcodec libjbig$<$<CONFIG:Debug>:d>)
+#install(TARGETS tstcodec RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
 
-add_executable (tstcodec85 "${CMAKE_CURRENT_SOURCE_DIR}/libjbig/tstcodec85.c")
-set_target_properties(tstcodec85 PROPERTIES DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX})
-target_link_libraries (tstcodec85 libjbig85$<$<CONFIG:Debug>:d>)
-install(TARGETS tstcodec85 RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+#add_executable (tstcodec85 "${CMAKE_CURRENT_SOURCE_DIR}/libjbig/tstcodec85.c")
+#set_target_properties(tstcodec85 PROPERTIES DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX})
+#target_link_libraries (tstcodec85 libjbig85$<$<CONFIG:Debug>:d>)
+#install(TARGETS tstcodec85 RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
 
 ########################################
 # pbm tools
 ########################################
 
-add_executable (pbmtojbg "${CMAKE_CURRENT_SOURCE_DIR}/pbmtools/jbgtopbm.c")
-set_target_properties(pbmtojbg PROPERTIES DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX})
-target_link_libraries (pbmtojbg libjbig$<$<CONFIG:Debug>:d>)
-install (TARGETS pbmtojbg RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+#add_executable (pbmtojbg "${CMAKE_CURRENT_SOURCE_DIR}/pbmtools/jbgtopbm.c")
+#set_target_properties(pbmtojbg PROPERTIES DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX})
+#target_link_libraries (pbmtojbg libjbig$<$<CONFIG:Debug>:d>)
+#install (TARGETS pbmtojbg RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
 
-add_executable (jbgtopbm85 "${CMAKE_CURRENT_SOURCE_DIR}/pbmtools/jbgtopbm85.c")
-set_target_properties(jbgtopbm85 PROPERTIES DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX})
-target_link_libraries (jbgtopbm85 libjbig85$<$<CONFIG:Debug>:d>)
-install (TARGETS jbgtopbm85 RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+#add_executable (jbgtopbm85 "${CMAKE_CURRENT_SOURCE_DIR}/pbmtools/jbgtopbm85.c")
+#set_target_properties(jbgtopbm85 PROPERTIES DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX})
+#target_link_libraries (jbgtopbm85 libjbig85$<$<CONFIG:Debug>:d>)
+#install (TARGETS jbgtopbm85 RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
 
-add_executable (jbgtopbm "${CMAKE_CURRENT_SOURCE_DIR}/pbmtools/jbgtopbm.c")
-set_target_properties(jbgtopbm PROPERTIES DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX})
-target_link_libraries (jbgtopbm libjbig$<$<CONFIG:Debug>:d>)
-install (TARGETS jbgtopbm RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+#add_executable (jbgtopbm "${CMAKE_CURRENT_SOURCE_DIR}/pbmtools/jbgtopbm.c")
+#set_target_properties(jbgtopbm PROPERTIES DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX})
+#target_link_libraries (jbgtopbm libjbig$<$<CONFIG:Debug>:d>)
+#install (TARGETS jbgtopbm RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
 
-add_executable (pbmtojbg85 "${CMAKE_CURRENT_SOURCE_DIR}/pbmtools/pbmtojbg85.c")
-set_target_properties(pbmtojbg85 PROPERTIES DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX})
-target_link_libraries (pbmtojbg85 libjbig85$<$<CONFIG:Debug>:d>)
-install (TARGETS pbmtojbg85 RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+#add_executable (pbmtojbg85 "${CMAKE_CURRENT_SOURCE_DIR}/pbmtools/pbmtojbg85.c")
+#set_target_properties(pbmtojbg85 PROPERTIES DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX})
+#target_link_libraries (pbmtojbg85 libjbig85$<$<CONFIG:Debug>:d>)
+#install (TARGETS pbmtojbg85 RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
