--- Portfile.old	2024-09-18 09:46:59.000000000 +0200
+++ Portfile	2024-09-18 13:06:14.000000000 +0200
@@ -670,7 +670,7 @@
     qtwebengine {
         {}
         "port:python27 port:py27-ply path:bin/ninja:ninja port:gperf port:bison port:flex port:qt5-qtwebengine-gn"
-        "port:fontconfig port:dbus path:lib/pkgconfig/harfbuzz.pc:harfbuzz path:lib/pkgconfig/glib-2.0.pc:glib2 port:zlib port:minizip port:libevent port:libxml2 path:lib/pkgconfig/poppler.pc:poppler port:pulseaudio path:lib/pkgconfig/icu-uc.pc:icu path:lib/libavcodec.dylib:ffmpeg port:libopus port:webp port:libpng port:lcms2 port:freetype port:re2 port:snappy"
+        "port:fontconfig port:dbus path:lib/pkgconfig/harfbuzz.pc:harfbuzz path:lib/pkgconfig/glib-2.0.pc:glib2 port:zlib port:minizip port:libevent port:libxml2 path:lib/pkgconfig/poppler.pc:poppler port:pulseaudio path:lib/pkgconfig/icu-uc.pc:icu port:libopus port:webp port:libpng port:lcms2 port:freetype port:re2 port:snappy"
         "qtdeclarative qtquickcontrols qtquickcontrols2 qtlocation qtwebchannel qttools"
         {"Qt WebEngine"}
         "very large and relatively new"
@@ -1796,16 +1796,15 @@
                 # see config_help.txt
                 configure.post_args-append         \
                     --                             \
-                    -webengine-native-spellchecker \
+                    -no-webengine-spellchecker \
                     -webengine-kerberos \
                     -webengine-proprietary-codecs \
-                    -system-webengine-icu \
-                    -system-webengine-ffmpeg
+                    -system-webengine-icu
 
                 # missing file spell_check_host_chrome_impl_mac.cc
                 # TODO: investigate further
                 configure.post_args-delete \
-                    -webengine-native-spellchecker
+                    -no-webengine-spellchecker
 
                 # GN bundled with qtwebengine-chromium has Qt-specific additions
                 # To avoid building the bundled GN from scratch when building qtwebengine,
