--- Portfile.old	2024-09-18 09:46:59.000000000 +0200
+++ Portfile	2024-09-18 13:06:14.000000000 +0200
@@ -670,7 +670,7 @@
     qtwebengine {
         {}
         "port:python27 port:py27-ply path:bin/ninja:ninja port:gperf port:bison port:flex port:qt5-qtwebengine-gn"
-        "port:fontconfig port:dbus path:lib/pkgconfig/harfbuzz.pc:harfbuzz path:lib/pkgconfig/glib-2.0.pc:glib2 port:zlib port:minizip port:libevent port:libxml2 path:lib/pkgconfig/poppler.pc:poppler port:pulseaudio path:lib/pkgconfig/icu-uc.pc:icu path:lib/libavcodec.dylib:ffmpeg port:libopus port:webp port:libpng port:lcms2 port:freetype port:re2 port:snappy"
+        "port:fontconfig port:dbus path:lib/pkgconfig/harfbuzz.pc:harfbuzz path:lib/pkgconfig/glib-2.0.pc:glib2 port:zlib port:minizip port:libevent port:libxml2 path:lib/pkgconfig/poppler.pc:poppler port:pulseaudio path:lib/pkgconfig/icu-uc.pc:icu port:libopus port:webp port:libpng port:lcms2 port:freetype port:re2 port:snappy"
         "qtdeclarative qtquickcontrols qtquickcontrols2 qtlocation qtwebchannel qttools"
         {"Qt WebEngine"}
         "very large and relatively new"
@@ -1778,6 +1778,8 @@
                 # do not opportunistically find MacPorts libraries (e.g. X11 libraries)
                 patchfiles-append patch-qtwebengine_tests.diff
 
+                patchfiles-append patch-qtwebengine_spellcheck.diff
+
                 # run syncqt.pl to generate headers needed to build from git snapshot (rather than release tarball)
                 # system Perl is sufficient, since syncqt.pl only uses standard core modules
                 # (although future macOS may not include Perl)
@@ -1796,16 +1796,15 @@
                 # see config_help.txt
                 configure.post_args-append         \
                     --                             \
-                    -webengine-native-spellchecker \
+                    -no-webengine-spellchecker \
                     -webengine-kerberos \
                     -webengine-proprietary-codecs \
-                    -system-webengine-icu \
-                    -system-webengine-ffmpeg
+                    -system-webengine-icu
 
