diff --git a/src/sycoca/ksycoca.cpp b/src/sycoca/ksycoca.cpp
index 02ed7bb4..aadc3ce1 100644
--- a/src/sycoca/ksycoca.cpp
+++ b/src/sycoca/ksycoca.cpp
@@ -240,7 +240,7 @@ bool KSycocaPrivate::openDatabase()
         static bool firstTime = true;
         if (firstTime) {
             firstTime = false;
-            if (KSandbox::isFlatpak()) {
+            /*if (KSandbox::isFlatpak())*/ {
                 // We're running inside flatpak, which sets all times to 1970
                 // So the first very time, don't use an existing database, recreate it
                 qCDebug(SYCOCA) << "flatpak detected, ignoring" << m_databasePath;
