diff --git a/src/QtAV_Global.cpp b/src/QtAV_Global.cpp
index c3374faa..c740fcf2 100644
--- a/src/QtAV_Global.cpp
+++ b/src/QtAV_Global.cpp
@@ -193,17 +193,8 @@ QString aboutQtAV_PlainText()
 QString aboutQtAV_HTML()
 {
     static QString about = QString::fromLatin1("<img src='qrc:/QtAV.svg'><h3>QtAV " QTAV_VERSION_STR_LONG "</h3>\n"
-            "<p>%1</p><p>%2</p><p>%3 </p>"
-            "<p>Copyright (C) 2012-2019 Wang Bin (aka. Lucas Wang) <a href='mailto:wbsecg1@gmail.com'>wbsecg1@gmail.com</a></p>\n"
-            "<p>%4: <a href='http://qtav.org/donate.html'>http://qtav.org/donate.html</a></p>\n"
-            "<p>%5: <a href='https://github.com/wang-bin/QtAV'>https://github.com/wang-bin/QtAV</a></p>\n"
-            "<p>%6: <a href='http://qtav.org'>http://qtav.org</a></p>"
-           ).arg(QObject::tr("Multimedia framework base on Qt and FFmpeg.\n"))
-            .arg(QObject::tr("Distributed under the terms of LGPLv2.1 or later.\n"))
-            .arg(QObject::tr("Shanghai, China"))
-            .arg(QObject::tr("Donate"))
-            .arg(QObject::tr("Source"))
-            .arg(QObject::tr("Home page"));
+            "<p>%1</p>"
+           ).arg(QObject::tr("Multimedia framework base on Qt and FFmpeg.\n"));
     return about;
 }
 
