diff --git a/widgets/QtAVWidgets/GLWidgetRenderer2.h b/widgets/QtAVWidgets/GLWidgetRenderer2.h
index ef282d03..e27a4277 100644
--- a/widgets/QtAVWidgets/GLWidgetRenderer2.h
+++ b/widgets/QtAVWidgets/GLWidgetRenderer2.h
@@ -22,6 +22,8 @@
 #ifndef QTAV_GLWIDGETRENDERER2_H
 #define QTAV_GLWIDGETRENDERER2_H
 
+#include <QtGlobal>
+
 #if !defined(QT_NO_OPENGL) && QT_VERSION < QT_VERSION_CHECK(6,0,0)
 #include <QtAVWidgets/global.h>
 #include <QtOpenGL/QGLWidget>
