--- Portfile.old	2024-09-10 03:59:43
+++ Portfile	2024-09-16 13:59:29
@@ -148,7 +148,7 @@
             8451800
         }
         ""
-        "path:lib/libavcodec.dylib:ffmpeg port:pulseaudio"
+        "path:lib/libavcodec.dylib:ffmpeg6 port:pulseaudio"
         "qtbase qtshadertools qtdeclarative qtquick3d"
         {"Qt Multimedia"}
         ""
@@ -223,7 +223,7 @@
             658112
         }
         ""
-        "port:gconf"
+        ""
         "qtbase qtdeclarative qtserialport"
         {"Qt Positioning"}
         ""
@@ -359,7 +359,7 @@
         }
         "path:bin/node:nodejs20 port:python${python_version} port:py${python_version}-html5lib port:py${python_version}-ply"
         "port:re2 path:lib/pkgconfig/icu-uc.pc:icu port:webp \
-            port:libopus path:lib/libavcodec.dylib:ffmpeg path:lib/pkgconfig/vpx.pc:libvpx \
+            port:libopus path:lib/libavcodec.dylib:ffmpeg6 path:lib/pkgconfig/vpx.pc:libvpx \
             port:snappy path:lib/pkgconfig/glib-2.0.pc:glib2 port:zlib port:minizip port:libevent \
             port:libxml2 port:libxslt port:lcms2 port:libpng path:include/turbojpeg.h:libjpeg-turbo \
             path:lib/pkgconfig/harfbuzz.pc:harfbuzz port:freetype port:pciutils port:pulseaudio"
